modules/gdscript/gdscript_parser.cpp

```diff
@@ -3731,87 +3825,21 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 	// This is called after the analyzer is done finding the type, so this should be set here.
 	DataType export_type = variable->get_datatype();
 
-	if (p_annotation->name == SNAME("@export")) {
-		if (variable->datatype_specifier == nullptr && variable->initializer == nullptr) {
-			push_error(R"(Cannot use simple "@export" annotation with variable without type or initializer, since type can't be inferred.)", p_annotation);
-			return false;
-		}
-
-		bool is_array = false;
-
-		if (export_type.builtin_type == Variant::ARRAY && export_type.has_container_element_type()) {
-			export_type = export_type.get_container_element_type(); // Use inner type for.
-			is_array = true;
-		}
-
-		if (export_type.is_variant() || export_type.has_no_type()) {
-			push_error(R"(Cannot use simple "@export" annotation because the type of the initialized value can't be inferred.)", p_annotation);
+	// Validate variable type with export.
+	if (!export_type.is_variant() && (export_type.kind != DataType::BUILTIN || export_type.builtin_type != t_type)) {
+		// Check for alternative or convertable types:
+		if (export_type.builtin_type == t_alternative_type) {
+			variable->export_info.type = t_alternative_type;
+		} else if (export_type.builtin_type == t_convertable_type) {
+			// convertable type (usually int <-> float) is ok
```

Comment:
Is this incomplete or can this else if be removed and replaced by a comment elsewhere?